core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/CoarseMesosSchedulerBackend.scala

```diff
@@ -308,6 +303,23 @@ private[spark] class CoarseMesosSchedulerBackend(
     }
   }
 
+    // ToDo: Abstract out each condition and log them.
+  def isOfferSatisfiesRequirements(slaveId: String, mem: Double, cpusOffered: Int,
```

Comment:
Also, a short scaladoc explaining the difference between `meetsConstraints` and `satisfiesRequirements`. To the casual reader they are the same.